shub/auth.py

```diff
@@ -0,0 +1,48 @@
+import os
+import netrc
+
+from os.path import expanduser
+
+from click import ClickException
+
+OS_WIN = True if os.name == 'nt' else False
+NETRC_FILE = expanduser('~/_netrc') if OS_WIN else expanduser('~/.netrc')
+
+
+def find_api_key():
+    """
+    Raises:
+        ClickException: if no credentials are found
+    """
+    key = os.getenv("SHUB_APIKEY")
+    if not key:
+        key = get_key_netrc()
+
+    if not key:
+        err = 'Not logged in. Please login first with: shub login'
+        raise ClickException(err)
+
+    return key
+
+
+def get_key_netrc():
+    """Gets the key from the netrc file"""
+    try:
+        info = netrc.netrc(NETRC_FILE)
+    except IOError:
+        return
+    try:
+        key, account, password = info.authenticators("scrapinghub.com")
+    except TypeError:
+        return
+    if key:
+        return key
+
+
+def write_key_netrc(key):
+    descriptor = os.open(
+        NETRC_FILE,
+        os.O_CREAT | os.O_RDWR | os.O_APPEND, 0o600)
+    with os.fdopen(descriptor, 'a+') as out:
+        line = 'machine scrapinghub.com login {0} password ""\n'.format(key)
```

Comment:
btw, looks like this is not valid .netrc entry and that confuses a lot. If you try the same approach for `storage.scrapinghub.com` (put key into login section and leave password with `""`) and use it with `curl -sn` or `httpie` you'll find out that `""` from password field are also encoded which produces incorrect basic auth token and leads to 401. 

what confuses me is that we are using .netrc not following it's spec thus I don't really understand why we use `.netrc` at all.